NixOS/nixpkgs
# process-cpp: Refactor & update

## Description of changes

- Add me to maintainers
- Move to by-name
- Switch to `finalAttrs` pattern
- Switch to `src.hash`
- Add `pkgConfigModules` & test
- Add `updateScript`
- unstable-2021-05-11 -> 3.0.1-unstable-2024-03-14
- Enable tests
- Enable `strictDeps`
- Add outputs

There's no changelog, but the [latest changes](https://gitlab.com/ubports/development/core/lib-cpp/process-cpp/-/commits/main/?ref_type=HEADS) are more or less just updating the C++ standard & fixing the tests on newer Linux. I think backporting this to 24.05 should be fine.

## Things done



- Built on platform(s)
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)



---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc
